merge redis offline engine from searx

return42 opened the pull request:
## What does this PR do?

merge redis offline engine from searx / for details read along the commit messages.

asciimoo committed:
[enh] add redis offline engine / https://redis.io/

Slightly modified merge of commit [97269be6], [01a8a5814a] and [c8d2b5eb] from
searx.

[97269be6] https://github.com/searx/searx/commit/97269be6
[01a8a581] https://github.com/searx/searx/commit/01a8a581
[c8d2b5eb] https://github.com/searx/searx/commit/c8d2b5eb

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

asciimoo committed:
[fix] correct kv template formatting and remove internal data

Slightly modified merge of commit [ea7ccf24] from searx.

[ea7ccf24] https://github.com/searx/searx/commit/ea7ccf24

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/engines/redis_server.py:
```diff
@@ -0,0 +1,75 @@
+result_template = 'key-value.html'
+exact_match_only = True
+
+redis_client = None
```
Should be `_redis_client` to avoid an error when the engine is loaded.

return42 replied:
Oops .. thanks! .. fixed in 39c18274

return42 committed:
[fix] enigine redis - avoid error when the engine is loaded

Should be _redis_client to avoid an error when the engine is loaded.

Suggested-by: @dalf https://github.com/searxng/searxng/pull/124#pullrequestreview-673885664
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
